src/main/java/org/broadinstitute/hellbender/tools/examples/ExampleReadWalkerWithVariantsSpark.java

```diff
@@ -0,0 +1,89 @@
+package org.broadinstitute.hellbender.tools.examples;
+
+import htsjdk.variant.variantcontext.VariantContext;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.apache.spark.api.java.function.Function;
+import org.broadinstitute.barclay.argparser.Argument;
+import org.broadinstitute.barclay.argparser.CommandLineProgramProperties;
+import org.broadinstitute.hellbender.cmdline.StandardArgumentDefinitions;
+import org.broadinstitute.hellbender.cmdline.programgroups.ExampleProgramGroup;
+import org.broadinstitute.hellbender.engine.FeatureContext;
+import org.broadinstitute.hellbender.engine.FeatureInput;
+import org.broadinstitute.hellbender.engine.ReferenceContext;
+import org.broadinstitute.hellbender.engine.spark.ReadWalkerContext;
+import org.broadinstitute.hellbender.engine.spark.ReadWalkerSpark;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+
+import java.util.List;
+
+/**
+ * Example/toy program that prints reads from the provided file or files along with overlapping variants
+ * (if a source of variants is provided). Intended to show how to implement the ReadWalker interface.
+ */
+@CommandLineProgramProperties(
+        summary = "Prints reads from the provided file(s) along with overlapping variants (if a source of variants is provided) to the specified output file (or STDOUT if none specified)",
+        oneLineSummary = "Print reads with overlapping variants",
+        programGroup = ExampleProgramGroup.class,
+        omitFromCommandLine = true
+)
+public final class ExampleReadWalkerWithVariantsSpark extends ReadWalkerSpark {
+    private static final long serialVersionUID = 1L;
+
+    @Argument(fullName = StandardArgumentDefinitions.VARIANT_LONG_NAME, shortName = StandardArgumentDefinitions.VARIANT_SHORT_NAME, doc = "One or more VCF files", optional = true)
+    private List<FeatureInput<VariantContext>> variants;
+
+    @Argument(fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME, shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME, doc = "Output file (if not provided, defaults to STDOUT)", common = false, optional = true)
+    private String outputFile = null;
+
+    @Argument(fullName = "groupVariantsBySource", shortName = "groupVariantsBySource", doc = "If true, group overlapping variants by their source when outputting them", optional = true)
```

Comment:
Its not particularly important but you don't need to specify the shot name here, it gets filled by barclay.